### PR TITLE
fix enable_xhost options battle btwn gnu and intel

### PR DESCRIPTION
**Pull request by loriab**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,9 +15,13 @@ list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
...
 option_with_default(CMAKE_BUILD_TYPE "Build type (Release or Debug)" Release)
-option_with_flags(ENABLE_XHOST "Enables processor-specific optimization" ON
-                  "-xHost" "-march=native")
-
+if(CMAKE_CXX_COMPILER_ID MATCHES Intel)
+option_with_flags(ENABLE_XHOST "Enables processor-specific optimization (with MSVC, it enables AVX2 instructions)" ON
+                  "-xHost" "-march=native" "/arch:AVX2")
+else()
+option_with_flags(ENABLE_XHOST "Enables processor-specific optimization (with MSVC, it enables AVX2 instructions)" ON
+                  "-march=native" "-xHost" "/arch:AVX2")
```
Comment (dgasmith) [inline CMakeLists.txt:23]: Order matters?

**Review by dgasmith — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (loriab) [reply CMakeLists.txt:23]: sadly, each compiler recognizes the other's xhost-like flag, though with different meanings.

Comment (loriab) [reply CMakeLists.txt:23]: net result is a language error! or no xhost activated. no idea why it's popping up lately, but Susi found it on psi4, so we fixed it there and libxc.

**Comment by dgasmith** [comment]
Travis isn't going to trigger due to their recent deprecations. I feel comfortable simply merging as Appveyor passed. Any opinions @loriab?

**Comment by loriab** [comment]
> Travis isn't going to trigger due to their recent deprecations. I feel comfortable simply merging as Appveyor passed. Any opinions @loriab?

Sorry, I missed above comment. Yes, I'm confident all changes in PR are >=harmless.

**Pull request merged**